perllib/FixMyStreet/Cobrand/Merton/Waste.pm

```diff
@@ -390,12 +390,20 @@ sub check_ggw_transfer_applicable {
     my ($old_garden) = grep { $_->{ServiceId} eq '409' } @$old_services;
     $old_garden->{transfer_uprn} = $details->{uprn};
 
-    return { error => 'no_previous' } unless ($old_garden && $old_garden->{ServiceTasks});
-    my $subscription_enddate = $old_garden->{ServiceTasks}->{ServiceTask}->{ServiceTaskSchedules}->{ServiceTaskSchedule}[0]->{EndDate}->{DateTime};
+    my $servicetask = $self->garden_current_service_from_service_units($old_services);
 
+    return { error => 'no_previous' } unless $servicetask;
+
+    my $subscription_enddate = _parse_schedules($servicetask)->{end_date};
     return { error => 'due_soon' } if ($subscription_enddate && $self->waste_sub_due($subscription_enddate));
 
+    my $old_subscription_bin_data = Integrations::Echo::force_arrayref($servicetask->{Data}, 'ExtensibleDatum');
+    my ($bin_number) = grep { $_->{DatatypeName} eq 'Quantity'} @{$old_subscription_bin_data->[0]->{ChildData}->{ExtensibleDatum}};
```

Comment:
You need to loop through `@$old_subscription_bin_data` here, I think - you do not know that the garden information will be in the first entry.